A/autoconf/patches/relocatable-autoconf.patch

```diff
@@ -0,0 +1,275 @@
+diff --git a/Makefile.in b/Makefile.in
```

Comment:
Any chance to upstream this patch? It's somewhat invasive, which makes it hard for us to maintain it

Reply:
It is.... truly invasive.  Line 10, where we're setting `M$ = /usr/bin/env m4` is particularly suspect.  I give low chance of this being upstreamed.  I adapted this from OpenWRT where they used to do this as well.  Note that it's not even truly relocatable without more environment variables added in as well: https://github.com/staticfloat/BB2.jl/blob/3df50b4ecb5fc15ee30e91ab21ab1c6a69cf529e/BinaryBuilderToolchains.jl/src/toolchains/HostToolsToolchain.jl#L108-L121

Reply:
Oh, ok, good to know we have a reference to look at, it'd be nice to have that as a comment for future reference 🙂

Reply:
https://gitlab.nic.cz/turris/openwrt/-/blob/80b865f1ab955df8012a3dacc56c30e3070bfc7c/tools/automake/patches/000-relocatable.patch